advanced_functionality/fastai_oxford_pets/fastai_lesson1_sagemaker_example.ipynb

```diff
@@ -0,0 +1,386 @@
+   "outputs": [],
+   "source": [
+    "path_anno = path/'annotations'\n",
+    "path_img = path/'images'\n",
+    "\n",
+    "fnames = get_image_files(path_img)\n",
+    "fnames[:5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload the data\n",
+    "We use the ```sagemaker.Session.upload_data``` function to upload our datasets to an S3 location. The return value inputs identifies the location -- we will use this later when we start the training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = sagemaker_session.upload_data(path=path, bucket=bucket, key_prefix=prefix)\n",
+    "print('input spec (in this case, just an S3 path): {}'.format(inputs))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Construct a script for training and inference\n",
+    "Here is the full code that both trains the model and does model inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!pygmentize source/pets.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Script Functions\n",
+    "\n",
+    "SageMaker invokes the main function defined within your training script for training. When deploying your trained model to an endpoint, the model_fn() is called to determine how to load your trained model. The model_fn() along with a few other functions list below are called to enable predictions on SageMaker.\n",
+    "\n",
+    "### [Predicting Functions](https://github.com/aws/sagemaker-pytorch-containers/blob/master/src/sagemaker_pytorch_container/serving.py)\n",
+    "* model_fn(model_dir) - loads your model.\n",
+    "* input_fn(serialized_input_data, content_type) - deserializes predictions to predict_fn.\n",
+    "* output_fn(prediction_output, accept) - serializes predictions from predict_fn.\n",
+    "* predict_fn(input_data, model) - calls a model on data deserialized in input_fn.\n",
+    "\n",
+    "The model_fn() is the only function that doesn't have a default implementation and is required by the user for using PyTorch on SageMaker. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a training job using the sagemaker.PyTorch estimator\n",
+    "\n",
+    "The `PyTorch` class allows us to run our training function on SageMaker. We need to configure it with our training script, an IAM role, the number of training instances, and the training instance type. \n",
+    "\n",
+    "For local training with GPU, we could set this to ```local_gpu```.  In this case, `instance_type` was set above based on your whether you're running a GPU instance. If `instance_type` is set to a SageMaker instance type (e.g. ml.p2.xlarge) then the training will happen on SageMaker.\n",
```

Comment:
[nit] In this case, `instance_type` was set ~above~ *below* based on ~your~ whether you're running a GPU instance

Reply:
Updated docs in notebook